### PR TITLE
Ref tracker TLC, makes unit test harddel errors more descriptive

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

The ref tracker now searches the last category of vars I know of that can cause issues, /static and /global vars.
Added a unit test for this behavior
Makes create_and_destroy actually attempt to search for refs when harddeling, which should save some coders time, and me suffering from having to explain my rats nest.
As a part of this, create_and_destroy now waits 30 minutes for the gc queue to clear, instead of 8. Hopefully this is enough.

## Why It's Good For The Game

Should make the operation of fixing harddels easier. Also finally completes our coverage, this'll help with things in future.

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
image
working, nice

File: code/modules/unit_tests/create_and_destroy.dm [hunk 1]
```diff
@@ -152,8 +152,8 @@
 			garbage_queue_processed = TRUE
 			break
 
-		if(world.time > start_time + time_needed + 8 MINUTES)
-			Fail("Something has gone horribly wrong, the garbage queue has been processing for well over 10 minutes. What the hell did you do")
+		if(world.time > start_time + time_needed + 30 MINUTES) //If this gets us gitbanned I'm going to laugh so hard
```
Comment (Mothblocks) [inline code/modules/unit_tests/create_and_destroy.dm:155]: 30 minute ci is very much not unheard of, you're fine

**Comment by Mothblocks** [comment]
So this will slow CI down a lot when something hard dels, that's fine.

How does this fare in a correct scenario, though? Where nothing hard dels? Still slower? How much?

**Comment by LemonInTheDark** [comment]
Shouldn't be slower at all, since the path I'm enabling is only called when something harddels. Outside of the new unit test of course

**Comment by Mothblocks** [comment]
:+1: Cool, just making sure

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**